dimod/bqm/adjdictbqm.py

```diff
@@ -652,3 +652,89 @@ def set_quadratic(self, u, v, bias):
 
         adj = self._adj
         adj[u][v] = adj[v][u] = bias
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __mul__(self, other):
+        if isinstance(other, (int, float)):
+            bqm = AdjDictBQM(vartype=self.vartype)
+            bqm.add_offset(self.offset * other)
+            for v, bias in self.linear.items():
+                bqm.add_variable(v, bias * other)
+
+            for (u, v), bias in self.quadratic.items():
+                bqm.add_interaction(u, v, bias * other)
+
+        elif isinstance(other, BQM):
+            if other.shape[1] > 0 or self.shape[1] > 0:
+                raise ValueError("Multiply is supported for BQMs with "
+                                 "linear terms only")
+            if other.vartype != self.vartype:
+                other = other.change_vartype(self.vartype)
+
+            bqm = AdjDictBQM(vartype=self.vartype)
```

Comment:
Again, I would probably start with a `bqm.copy()`, though at the very least should do `bqm = type(self)(vartype=self.vartype)` to accommodate subclasses. Applies other places.

Reply:
for the product of two `bqm`s, I think making a copy makes it convoluted, I basically have to replace every bias.